v3.0.py

```diff
@@ -0,0 +1,295 @@
+"""This is an interactive data visualization map that allows users to determine the most common spoken languages for countries around the world.
+TO BEGIN: Press on any key, and then click on any of the points. If you would like to see the most common language for that specifc country, click on its
+centroid and click on any key other than 2 and 3. For second and third most common languages, click on 2 and 3.
+
+PSA: Our files kept getting corrupted once we pulled them from git. If this happens to you then email us and we will send you the correct file via email.
+Have fun and thank you!  """
+
+import csv
+import pygame
+import pandas as pd
+import numpy as np
+import xlrd
+import math
+
+#def processing_data():
+#reading the data and turning the country and language (either most spoken, second most spoken or third most spoken) into a dictionary
+my_dic_first_language = pd.read_excel('data2.xlsx', index_col=0, parse_cols = "A:B").to_dict()
+my_dic_second_language = pd.read_excel('data2.xlsx', index_col=0, parse_cols = "A,C").to_dict()
+my_dic_third_language = pd.read_excel('data2.xlsx', index_col=0, parse_cols = "A,D").to_dict()
+
+#finding value of the these values in order to reach the dictionary, the dictionary was returned within a dictionary
+a = my_dic_first_language['Dari Persian']
+b = my_dic_second_language['Pashtu']
+c = my_dic_third_language['Turkic']
+
+#reading the data for longitude and latitude for each country from a different source
+country = pd.read_excel(open('bleh.xlsx', 'rb'), parse_cols = "A")
+position = pd.read_excel('bleh.xlsx', index_col=0, parse_cols = "A,D").to_dict()
+position2 = pd.read_excel('bleh.xlsx', index_col=0, parse_cols = "A,E").to_dict()
+
+d = position['LAT2']
+h = position2['LONG2']
+
+#creates list of the first column (or the countries) for both sources
+ps = pd.read_excel('data2.xlsx')
+cd = pd.read_excel('bleh.xlsx')
+
+#compares both lists of countries and appends a list of the common countries
+real_countries = []
+for i in cd.iloc[:, 0].tolist():
+    if i in ps.iloc[:, 0].tolist():
+        real_countries.append(i)
+for i in ps.iloc[:, 0].tolist():
+    if i in cd.iloc[:, 0].tolist():
+        real_countries.append(i)
+
+#creates five new dictionaries for the three languages and the position as values and countries as keys
+new_first = {}
+new_second = {}
+new_third = {}
+new_pos = {}
+new_pos2 = {}
+
+#a function that returns a dictionary of values of a dictionary
+def common_dicts(dict1, dict2, lst):
+    for k, v in dict1.items():
+        for i in lst:
+            if i == k:
+                dict2[k] = v
+    return dict2
+
+#crreates a dictionary of keys that were in the countries list
+new1= common_dicts(a,new_first,real_countries)
```

Comment:
Instead of putting these into an array after assigning them to the variables, initialize them as part of the array. There's less confusion that way.